be/src/util/string_parser.hpp

```diff
@@ -253,7 +253,7 @@ T StringParser::string_to_int_internal(const char* s, int len, ParseResult* resu
     switch (*s) {
     case '-':
         negative = true;
-        max_val = StringParser::numeric_limits<T>(false) + 1;
+        max_val = StringParser::numeric_limits<T>(false) + UnsignedT(1);
```

Comment:
maybe we should split `max_val` & `min_val` usage? the UB occurs when T is a signed number. in that case, this is always UB.